pyispyb/config.py

```diff
@@ -34,23 +34,16 @@
 PROJECT_ROOT = os.path.abspath(os.path.dirname(__file__))
 RESOURCES_ROOT = os.path.join(PROJECT_ROOT, "resources")
 
-yaml_settings = dict()
-AUTH_CONFIG = os.path.realpath(
-    os.path.join(PROJECT_ROOT, "..", os.getenv("ISPYB_AUTH", "auth.yml"))
-)
-try:
-    with open(AUTH_CONFIG) as f:
-        yaml_settings.update(yaml.safe_load(f))
-except IOError:
-    raise Exception(f"Could not access auth config: {AUTH_CONFIG}")
-
 
-def get_env(name: str):
-    res = os.getenv(name, None)
+def get_env_file():
+    res = os.getenv("ISPYB_ENVIRONMENT", None)
     if res is None or res == "":
-        raise Exception(f"You must define env variable {name}")
-    else:
-        return res
+        return "config/.env"
+    v = f"config/{res}.env"
+    if os.path.exists(v):
+        return v
+
+    raise Exception(f"Config file {v} could not be found.")
```

Comment:
I'm not sure if an .env file should be a requirement, as its just an additional way to set the env vars. If you're running in docker for example you might just pass them through, rather than rely on a file?

pydantic will still catch if required env vars are missing

Reply:
That is only raised if you have set up `ISPYB_ENVIRONMENT` to some value and the corresponding env file does not exist.
If you have not defined it, it just uses the default `.env` file which provides some defaults but is incomplete and still relies on some env vars being defined.